### PR TITLE
Fix date format issue in transaction count retrieval

### PR DESCRIPTION
**Pull request by arunans23**
Fix date format issue in transaction count retrieval

### CODE REVIEW

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java [hunk 1]
```diff
@@ -112,11 +112,12 @@ public synchronized void addTransaction() throws TransactionCounterException {
      * @throws TransactionCounterException Error occurred while fetching data from the database.
      */
     public long getTransactionCountOfMonth(int year, int monthNumber) throws TransactionCounterException {
-
-        String dateString = year + "-" + monthNumber + "-01";
+        String monthNumberStr = Integer.toString(monthNumber);
+        monthNumberStr = ((monthNumberStr).length() == 1) ? "0" + monthNumberStr : monthNumberStr;
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java:116]: shall we remove all the enclosing brackets .
or at least the one around (monthNumberStr)

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java [hunk 2]
```diff
@@ -149,8 +150,8 @@ public long getTransactionCountOfMonth(int year, int monthNumber) throws Transac
         try (Connection dbConnection = getConnection();
              PreparedStatement prepStmt = dbConnection.prepareStatement(
                      TransactionQueryHelper.GET_TRAN_COUNT_DATA_FOR_A_TIME_PERIOD)) {
-            prepStmt.setString(1, startDate);
-            prepStmt.setString(2, endDate);
+            prepStmt.setDate(1,  new java.sql.Date(ConverterUtil.convertToDate(startDate).getTime()));
+            prepStmt.setDate(2, new java.sql.Date(ConverterUtil.convertToDate(endDate).getTime()));
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java:154]: shall we add this new java.sql.Date(ConverterUtil.convertToDate(endDate).getTime()) to a function and re use ( a signle line function .. ) use it in line 238 as well.

**Comment by Nirothipan** [comment]
Also , are all the db queries covered by the test case ?  do we have no issues while retrieving result set in line 160

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java (hunk 1 above)
Comment (arunans23) [reply components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java]: Fixed

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java (hunk 2 above)
Comment (arunans23) [reply components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java]: Fixed

**Comment by arunans23** [comment]
> Also , are all the db queries covered by the test case ? do we have no issues while retrieving result set in line 160
> 
> while (rs.next()) {
> data.add(new String[]{rs.getString(1), rs.getString(2),
> String.valueOf(rs.getLong(3)), rs.getString(4)});
> }

All DBQueries are covered.

**Pull request merged**